### PR TITLE
jewel: rgw: remove unnecessary 'error in read_id for object name: default'

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/20007

### CODE REVIEW

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -385,7 +385,9 @@ int RGWSystemMetaObj::init(CephContext *_cct, RGWRados *_store, bool setup_obj,
     } else if (!old_format) {
       r = read_id(name, id);
       if (r < 0) {
-        ldout(cct, 0) << "error in read_id for object name: " << name << " : " << cpp_strerror(-r) << dendl;
+        if (r != -ENOENT) {
```
Comment (joscollin) [inline src/rgw/rgw_rados.cc]: Can you combine the if conditions ?

Comment (smithfarm) [inline src/rgw/rgw_rados.cc]: Hi Jos - We generally don't "clean up" master commits when backporting, preferring to take them verbatim unless doing so causes a serious error.

**Comment by smithfarm** [comment]
Thanks for adding the "needs-qa" label. (That said, backport PRs in general are not supposed to ever get merged until they pass integration testing and explicit approval is obtained from the relevant lead.)

File: src/rgw/rgw_rados.cc (hunk 1 above)
Comment (joscollin) [inline src/rgw/rgw_rados.cc]: I wasn't aware of that. Thanks for the info.

**Comment by yuriw** [comment]
@smithfarm this gave me huge list of conflicts, pls rebase

**Comment by smithfarm** [comment]
@yuriw This is a jewel PR.

**Comment by smithfarm** [comment]
@oritwas This passed an rgw suite at http://tracker.ceph.com/issues/20613#note-8

Please review.

**Pull request merged**